File: java/src/main/java/com/penapereira/patterns/visitor/ConcreteElementA.java
package com.penapereira.patterns.visitor;

/** Element A; accepts a visitor and exposes operationA to it. */
public final class ConcreteElementA implements Element {

    @Override
    public void accept(Visitor visitor) {
        visitor.visitConcreteElementA(this);
    }

    public String operationA() {
        return "A";
    }
}

File: javascript/src/visitor/concrete-element-a.js
// Visitor: define new operations over an object structure without changing its classes.
// A visitor is any object with visitConcreteElementA/B; an element any object with accept().

export class ConcreteElementA {
  accept(visitor) {
    visitor.visitConcreteElementA(this);
  }

  operationA() {
    return "A";
  }
}
